.eslintrc.js

```diff
@@ -0,0 +1,114 @@
+/* [object Object]*/ module.exports = {
+    root: true,
+    parser: '@typescript-eslint/parser',
+    parserOptions: {
+        ecmaVersion: 'latest',
+        sourceType: 'module',
+        requireConfigFile: false,
+    },
+    env: {
+        node: true,
+        mocha: true,
+        es2024: true,
+    },
+    plugins: ['@typescript-eslint', '@stylistic', 'unicorn', 'header', 'security-node'],
+    extends: [
+        'eslint:recommended',
+        'plugin:@typescript-eslint/eslint-recommended',
+        'plugin:@typescript-eslint/recommended',
+        'plugin:prettier/recommended',
+    ],
+    rules: {
+        curly: 2, // Enforce braces on "if"/"for"/etc.
+        'id-length': [
+            'error',
+            {
+                min: 1,
+                max: 40,
+                exceptionPatterns: [
+                    '^INSERT_TO_CURSOR_POSITION_NOTIFICATION_METHOD$',
+                    '^DID_CHANGE_DEPENDENCY_PATHS_NOTIFICATION_METHOD$',
+                ],
+            },
+        ],
+        // https://typescript-eslint.io/rules/naming-convention/
+        '@typescript-eslint/naming-convention': [
+            'error',
+            {
+                selector: 'default',
+                format: ['camelCase', 'PascalCase'],
+                // Allow underscores.
+                leadingUnderscore: 'allowSingleOrDouble',
+                trailingUnderscore: 'allowSingleOrDouble',
+            },
+            // Allow object properties, enums, and methods to have any format
+            {
+                selector: ['objectLiteralProperty', 'classMethod', 'enumMember'],
+                format: null,
+            },
+            // Allow constants to use UPPER_CASE
+            {
+                selector: 'variable',
+                modifiers: ['const'],
+                format: ['camelCase', 'PascalCase', 'UPPER_CASE'],
+            },
+        ],
+        // Avoid accidental use of "==" instead of "===".
+        eqeqeq: 'error',
+        '@typescript-eslint/no-namespace': 'error',
+        '@typescript-eslint/no-floating-promises': 'off', // Temporarily disabled
```

Comment:
I would prefer this enabled - it allows to catch when people forget to await.

Reply:
Definitely valuable to enable this, but if the codebase isn't prepared for it, it may trigger a lot of warnings. Worth a try though.

Reply:
should be pretty simple to enable it in runtimes